portfolio/src/main/webapp/CreatePuzzlePage.js

```diff
@@ -0,0 +1,49 @@
+const createPuzzleForm = document.getElementById("create-puzzle-form");
+const responseMessage = document.getElementById("upload-message");
+
+//Handles the Create Puzzle Form Submission
+const handleCreatePuzzleFormSubmission = (e) =>{
+    e.preventDefault();
+    createPuzzleForm.reset();
+
+    //Getting the link to the newly created puzzle
+    getPuzzleLink();
+}
+
+async function getPuzzleLink () {
+    try{
+        const response = await fetch("/upload");
+        const text = await response.json();
```

Comment:
Heads up. In @Alexjoshua14  #12 this will return the entire puzzle object instead of just a message. So you'll have to construct the URL using the puzzle id. I think the url could just be:
```javascript
const puzzle = await response.json();
// This assumes viewpuzzlepage.html exists
const viewPuzzlePageUrl = `/viewpuzzlepage.html?id={puzzle.puzzleId}`;
```